src/main/java/com/datatorrent/netlet/util/CircularBuffer.java

```diff
@@ -52,13 +56,14 @@
   @SuppressWarnings("unchecked")
   public CircularBuffer(int n, int spin)
   {
-    int i = 1;
-    while (i < n) {
-      i <<= 1;
-    }
+    if (n < 0 || n > MAX_SIZE) throw new RuntimeException(String.format("Error: value too large: %d%n", n));
+
+    final int size = (0 == n) ? 1
+            : (0 == (n & (n - 1))) ? n
+            : (1 << (32 - Integer.numberOfLeadingZeros(n)));
 
-    buffer = (T[])new Object[i];
-    buffermask = i - 1;
+    buffer = (T[])new Object[size];
```

Comment:
Call to another constructor must be first statement of function body, so cannot do this.